Sorry for the big commit

Pop000100 opened the pull request:
- Add support for Rod Of Discord Hotkey mod for Rod of Warping.
- fixed the ninja gear's dash not working.
- fixed Reinforced Vest and company
- fixed the error due to the entity source changes
- fixed the crashes with the new versions of TModloader
- made Items reasearchable

Pop000100 committed:
Sorry for the big commit

- Add support for Rod Of Discord Hotkey mod for Rod of Warping.
- fixed the ninja gear's dash not working.
- fixed Reinforced Vest and company
- fixed the error due to the entity source changes
- fixed the crashes with the new versions of TModloader
- made Items reasearchable

Pop000100 committed:
Increment Version Number

ditto.

Pop000100 committed:
Unguard the autofisher

- The autofisher setup needs to run all the time so that if or when the Mod.Call happens it doesn't error

Pop000100 committed:
Increment the version

Gah Forget to increment the version on last commit

gardenappl commented on Items/Weapons/TopazStaff.cs:
```diff
@@ -3,7 +3,7 @@
 using Terraria;
 using Terraria.ID;
-using Terraria.Localization;
+using Terraria.GameContent.Creative;
```
Did you forget to make this researchable?

Pop000100 replied:
Yes, yes I did. fixed in the next commit.

gardenappl commented on MiscUtils.cs:
```diff
@@ -266,6 +265,11 @@ public static bool ChestHasItem(Chest shop, int type)
 			return false;
 		}
 
+		public static bool IsTheSameAs(Item item, Item compareItem)
```
Honestly I'm not sure what's the difference between this and just `item1.type == item2.type`. I think I initially used IsTheSameAs just because I copy-pasted that code from somewhere else, but I'm not sure if I like it.

Pop000100 replied:
I am going to remove it and test it for a bit, it looks like the code in 1.4 calls it so it might be redundant now anyway.

Update: Tested it on a server and singleplayer nothing seems broken

Pop000100 committed:
Requested changes + 1

- Remove a log that I left while debugging
- Fix the Topaz staff
- Simplify the chest vacuum

gardenappl commented:
Thanks
